### ISSUE TITLE
Number to text conversion is incorrect (it's coming with scientific notation)

### ISSUE THREAD

**Issue by alexandresilvawek** [MonetDB/MonetDB#7258]
**Describe the bug**
I have a field of type double and I want to convert its content to text.
But, the result of the conversion seems incorrect to me (it's coming with scientific notation).

**To Reproduce**
Just trying to convert the content of a field of type double to text.
I think for scientific notation to show up, the content must have a zero at the end (as can be seen in screenshot below).

**Expected behavior**
Convert the contents of the double field to text only (no scientific notation).

**Screenshots**
![image](https://user-images.githubusercontent.com/86444458/155986046-61d2294a-f04c-4d47-aec6-b9a17477057e.png)

**Software versions**
 - MonetDB version number: 11.43.9
 - OS and version: Windows 10
 - Installed from release package or self-installed and compiled: release package

**Comment by sjoerdmullender** [comment]
I see nothing wrong here.  If you don't want scientific notation, you should consider using DECIMAL instead of FLOAT/DOUBLE.

**Comment by alexandresilvawek** [comment]
Hi @sjoerdmullender ,

I understood.
Thank you for the explanation.
But unfortunately I have read-only access to this data.

Well, I tried to meet my need in another way, but it seems that there is another problem.

See the screenshots below:

First:
![1](https://user-images.githubusercontent.com/86444458/156026434-edcd4c10-48b0-4bb8-8453-644b9c0a3d3f.png)

Second:
![2](https://user-images.githubusercontent.com/86444458/156026456-3abd26d9-9be5-48b5-ab43-11c637690a00.png)

Third:
![3](https://user-images.githubusercontent.com/86444458/156026520-4d9c0e39-2f44-4317-a307-e11909a92f4a.png)

Fourth:
![4](https://user-images.githubusercontent.com/86444458/156026577-c3fad8a8-2b54-4316-847c-0c4f67813517.png)

I believe that:
- The first and second should have the same result;
- The third and fourth should have the same result;

I don't understand why the results were different.

Could you help me with this?

**Comment by sjoerdmullender** [comment]
The problem is that the scaling of a SQL REAL value (32 bit floating point) to a DECIMAL (an integer with a scale that says where the decimal point should be) is done in too low a precision.  Scaling involves a multiplication with a power of 10.  You can work around this by doing an extra cast to DOUBLE (64 bit floating point):
`cast(cast(nr_atendimento as double) as decimal(18,11))`

**Comment by alexandresilvawek** [comment]
Hi @sjoerdmullender ,

I tested it with the extra cast to double, but unfortunately it didn't work.
It gave the same results as before.

Do you have any other ideas to help me?

**Comment by sjoerdmullender** [comment]
For me it works:
```
sql>select r from t;
+-----------------+
| r               |
+=================+
| 7.9025e+05      |
| 4.23862e+06     |
| 6.72407e+06     |
+-----------------+
3 tuples
sql>select cast(r as decimal(18,11)) from t;
+----------------------+
| %1                   |
+======================+
|   790250.01654648832 |
|  4238620.07385554944 |
|  6724069.54441703424 |
+----------------------+
3 tuples
sql>select cast(cast(r as double) as decimal(18,11)) from t;
+----------------------+
| %2                   |
+======================+
|   790250.00000000000 |
|  4238620.00000000000 |
|  6724070.00000000000 |
+----------------------+
3 tuples
```
So, what did you do differently?

**Comment by alexandresilvawek** [comment]
See how I did it here:

1:
![image](https://user-images.githubusercontent.com/86444458/156174810-201d67fb-27e4-4f15-b5b4-09aa807e02f8.png)

2:
![image](https://user-images.githubusercontent.com/86444458/156174896-b914fc83-2957-49dc-b4d4-a68661aa09ec.png)

3:
![image](https://user-images.githubusercontent.com/86444458/156174973-f7db598e-e807-4d86-990b-6834f26cc4f6.png)

4:
![image](https://user-images.githubusercontent.com/86444458/156175038-9321c963-5a3f-4472-a4a7-5ae6b2293b14.png)

If it helps you, I can send the script to create the same table and the same data I have here.

**Comment by PedroTadim** [comment]
I am closing this issue for now.

**Comment by alexandresilvawek** [comment]
Hello Pedro, hello Sjoerd,

From what I understand, the change made by Sjoerd was released in version 11.43.13 and should help or maybe even solve this problem I found.

I just did the same tests on version 11.43.13 and the problem is still the same.

Could you please check this case again?
